File: backend/app/schemas/application.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class ApplicationCreate(BaseModel):
    job_id: UUID
    cover_letter: Optional[str] = None
    form_answers: Optional[dict] = None


class ApplicationStatusUpdate(BaseModel):
    status: str
    notes: Optional[str] = None


class ApplicationResponse(BaseModel):
    id: UUID
    job_id: UUID
    status: str
    cover_letter: Optional[str] = None
    form_answers: Optional[dict] = None
    automation_log: Optional[dict] = None
    submitted_at: Optional[datetime] = None
    error_message: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    # Joined from Job
    company_name: Optional[str] = None
    job_title: Optional[str] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/application.ts
export interface Application {
  id: string;
  job_id: string;
  tailored_resume_id: string | null;
  status: string;
  cover_letter: string | null;
  form_answers: Record<string, unknown> | null;
  automation_log: LogEntry[] | null;
  submitted_at: string | null;
  error_message: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
  // Joined from Job
  company_name: string | null;
  job_title: string | null;
}

export interface LogEntry {
  step: number;
  action: string;
  timestamp: string;
  success?: boolean;
  details?: string;
}

export interface RecentApplication {
  id: string;
  status: string;
  created_at: string;
  company_name: string | null;
  job_title: string | null;
}

export interface DashboardStats {
  total_applications: number;
  by_status: Record<string, number>;
  this_week: number;
  this_month: number;
  response_rate: number;
}
